### PR TITLE
SWC-5702: reset empty fav UI on area change

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java [hunk 1]
```diff
@@ -326,6 +326,7 @@ public void getMoreProjects() {
 		if (isOwner) {
 			view.setLastActivityOnColumnVisible(true);
 			view.showProjectFiltersUI();
+			view.setFavoritesHelpPanelVisible(false);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java:329]: This fixes the bug that was reported.

File: src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java [hunk 2]
```diff
@@ -601,24 +602,21 @@ public void getFavorites() {
 			public void onSuccess(List<EntityHeader> result) {
 				if (filterType == ProjectFilterEnum.FAVORITES) {
 					// convert to Project Headers
-					if (result.size() == 0) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java:604]: Show the FavoritesHelpPanel if no _Projects_ have been marked as being a favorite.

File: src/main/resources/org/sagebionetworks/web/client/view/ProfileViewImpl.ui.xml [hunk 3]
```diff
@@ -98,7 +98,7 @@
 													addStyleNames="font-size-90 favoriteIcon" />
 												<b:Heading text="Favorites" size="H4"></b:Heading>
 												<bh:Text
-													text="Save the entity you’d like to revisit by marking it as a favorite. To mark an entity as a favorite, click the star icon next to its name."></bh:Text>
+													text="Save the Project you’d like to revisit by marking it as a favorite. To mark an Project as a favorite, click the star icon next to its name."></bh:Text>
```
Comment (jay-hodgson) [inline src/main/resources/org/sagebionetworks/web/client/view/ProfileViewImpl.ui.xml:101]: To support SWC-5451

Comment (jay-hodgson) [reply src/main/resources/org/sagebionetworks/web/client/view/ProfileViewImpl.ui.xml:101]: In this context, we only show Projects

**Review by nickgros — APPROVED** [review]

**Pull request merged**